src/components/TodoFilter/TodoFilter.tsx

```diff
@@ -1,34 +1,96 @@
-export const TodoFilter = () => (
-  <form className="field has-addons">
-    <p className="control">
-      <span className="select">
-        <select data-cy="statusSelect">
-          <option value="all">All</option>
-          <option value="active">Active</option>
-          <option value="completed">Completed</option>
-        </select>
-      </span>
-    </p>
-
-    <p className="control is-expanded has-icons-left has-icons-right">
-      <input
-        data-cy="searchInput"
-        type="text"
-        className="input"
-        placeholder="Search..."
-      />
-      <span className="icon is-left">
-        <i className="fas fa-magnifying-glass" />
-      </span>
-
-      <span className="icon is-right" style={{ pointerEvents: 'all' }}>
-        {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
-        <button
-          data-cy="clearSearchButton"
-          type="button"
-          className="delete"
+import React, { useEffect, useState } from 'react';
+import { getTodos } from '../../api';
+import { Todo } from '../../types/Todo';
+
+type Props = {
+  setTodos: (a: Todo[]) => void
+};
+
+export const TodoFilter: React.FC<Props> = ({ setTodos }) => {
+  const [title, setTitle] = useState('');
+
+  useEffect(() => {
+    const filteredByTitle = (todos: Todo[]) => {
```

Comment:
The function 'filteredByTitle' is defined inside a useEffect hook, which is not a recommended practice. Move this function outside of the useEffect hook.